### ISSUE TITLE
Breaks on reload

### ISSUE THREAD

**Issue by PurelyAliyah** [lucko/LuckPerms#1163]
I love this plugin but I also love the fuction to reload and with the permissions plugin being the base of everything a server is I would really like for me to be able to reload without luckperms unsyncing and forgetting that it's a permissions plugin.

**Comment by PurelyAliyah** [comment]
Also Forgot I should specify that I'm using /reload not /lp reloadconfig, And yes I know /reload can break things- but It's a lot better than restarting everytime I'm working on a plugin config change, most plugins don't have a built in reload command.

**Comment by Andre601** [comment]
> And yes I know /reload can break things- but It's a lot better than restarting everytime I'm working on a plugin config change, most plugins don't have a built in reload command.

Nope it isn't.
At NO circumstance AT ALL should you use a /reload over a /restart
Plugins (specially perm-plugins) can and WILL break on reloads.
So you have to go to the pain of restarting the server.

**Comment by BrainStone** [comment]
Also when making a plugin the config reload is among the first things I implement. So I honestly think you should nag the authors of said plugins to add that functionality. Also keeping in mind that chances are that such plugins will break silently too.

**Comment by lucko** [comment]
I agree with what has been said above, but still consider this to be a bug. 

What sort of errors do you get on /reload? Any error messages in console?

**Comment by PurelyAliyah** [comment]
It doesn't say anything in console, It just stops working.

**Comment by PurelyAliyah** [comment]
> At NO circumstance AT ALL should you use a /reload over a /restart
Plugins (specially perm-plugins) can and WILL break on reloads.
So you have to go to the pain of restarting the server.

I don't like losing players because I restart too often, So I disagree. I'm not here to argue. This is an issue thread, not for you to tell me how to run my server. I've been doing this for five years, I know what reloading can do to a server- And up until this point I haven't had any problems with it.

**Comment by BrainStone** [comment]
Sure while this is an issue thread, it’s an issue caused by not following best practices and ignoring warnings that whatever you’re doing might break stuff.

Using a reload is like playing Russian Roulette with your foot. It certainly won’t kill you. And most of the time you’ll be fine. But eventually you’ll blow your foot off. And it’s just the appropriate response to point out that you shouldn’t do that.  
Keep in mind we all run servers. And we all know the stupidity of players (“server need restart? Bad server. Not come back”) but there’s a reason essentially everyone warns highly against reloading. Paper even forces you to confirm after it explicitly warns you. And there have even been discussions to completely remove the reload because most plugin DEVs can’t write a plugin that doesn’t break on reload. It’s not because they’re bad. It’s because it’s hard.

And as it seems I was right. It appears that another plugin was silently breaking in the background. You just never noticed it before. Or it only breaks with LP being present (for whatever reason)

So in summary we’re not telling you not to use reload because we don’t know what it’s useful for and what it’s intended to do (and that it’s main purpose is to prevent having to restart the server), but because we know that you virtually can’t know if a plugin was written by a 12-year-old that started programming 2 days ago or a 50 year old programming veteran that has has 60 years of experience and started using Java the day it came out.

**Comment by lucko** [comment]
Please re-open when you have an error message I can look into. :)
